src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationRefresherProvider.cs

```diff
@@ -23,12 +23,6 @@ public AzureAppConfigurationRefresherProvider(IConfiguration configuration, ILog
                 {
                     if (provider is IConfigurationRefresher refresher)
                     {
-                        // Use _loggerFactory only if LoggerFactory hasn't been set in AzureAppConfigurationOptions
```

Comment:
This is where we get the logger factory from DI and assign it to the `IConfigurationRefresher` refresher, which is actually an instance of `AzureAppConfigurationProvider`. 
If we are deprecating LoggerFactory from `IConfigurationRefresher`, we would need to cast this `provider` to `AzureAppConfigurationProvider` and assign `_loggerFactory` obtained from DI to the concrete instance of `AzureAppConfigurationProvider`. Otherwise, `AzureAppConfigurationProvider.LoggerFactory` will always be null.

```
if (provider is AzureAppConfigurationProvider appConfigurationProvider)
{
    appConfigurationProvider.LoggerFactory = _loggerFactory;                 
    refreshers.Add(appConfigurationProvider);
}
```

@jimmyca15, do you see an issue with casting to `AzureAppConfigurationProvider` instead of `IConfigurationRefresher`?

Reply:
No. I don't see a problem.